utils/frame/remote-externalities/src/lib.rs

```diff
@@ -333,16 +334,28 @@ impl<B: BlockT> Builder<B> {
 		info!(target: LOG_TARGET, "Querying a total of {} keys", keys.len());
 
 		let mut key_values: Vec<KeyPair> = vec![];
-		for key in keys {
-			let value =
-				RpcApi::<B>::get_storage(self.as_online().rpc_client(), key.clone(), Some(at))
-					.await
-					.map_err(|e| {
-						error!(target: LOG_TARGET, "Error = {:?}", e);
-						"rpc get_storage failed"
-					})?;
-			key_values.push((key, value));
-			if key_values.len() % 1000 == 0 {
+		let client = self.as_online().rpc_client();
+		for chunk_keys in keys.chunks(BATCH_SIZE) {
+			let batch = chunk_keys
+				.iter()
+				.cloned()
+				.map(|key| {
+					assert!(key.0.len() == 32);
+					(
+						"state_getStorage",
+						JsonRpcParams::Array(vec![serde_json::to_value(key).unwrap()]),
```

Comment:
why are you not specifying `Some(at)` here anymore?

Reply:
@emostov it's possible to specify at which block to get the state which at meant by `Some(at)`, see https://github.com/paritytech/substrate/blob/master/client/rpc-api/src/state/mod.rs#L68

Reply:
Thanks - I was just wondering if the omission was intentional on Kian's part

Reply:
ahh damn, no this was a mistake. added now.